Sources/Wendy/cli/commands/DiscoverCommand.swift

```diff
@@ -101,26 +49,66 @@ struct DiscoverCommand: AsyncParsableCommand {
 
         switch type {
         case .usb:
-            usbDevices = await discovery.findUSBDevices()
+            usbDevices = try await Noora().progressStep(
+                message: "Discovering Wendy USB devices",
+                successMessage: nil,
+                errorMessage: nil,
+                showSpinner: !json
+            ) { progress in
+                async let _usbDevices = await discovery.findUSBDevices()
+                let usb = await _usbDevices
+                return usb
+            }
             logDevicesFound(usbDevices, deviceType: "USB device(s)", logger: logger)
 
         case .ethernet:
-            ethernetDevices = await discovery.findEthernetInterfaces()
+            ethernetDevices = try await Noora().progressStep(
+                message: "Discovering Wendy Ethernet interfaces",
+                successMessage: nil,
+                errorMessage: nil,
+                showSpinner: !json
+            ) { progress in
+                async let _ethernetDevices = await discovery.findEthernetInterfaces()
+                let ethernet = await _ethernetDevices
+                return ethernet
+            }
             logDevicesFound(ethernetDevices, deviceType: "Ethernet interface(s)", logger: logger)
 
         case .lan:
-            lanDevices = try await discovery.findLANDevices()
+            lanDevices = try await Noora().progressStep(
+                message: "Discovering Wendy LAN devices",
+                successMessage: nil,
+                errorMessage: nil,
+                showSpinner: !json
+            ) { progress in
```

Comment:
We can't show the progress view at all when `json == true`. Otherwise our stdout is not pure JSON